proxy/src/main/java/com/velocitypowered/proxy/command/VelocityCommandManager.java

```diff
@@ -155,19 +154,18 @@ public boolean executeImmediately(final CommandSource source, final String cmdLi
     Preconditions.checkNotNull(source, "source");
     Preconditions.checkNotNull(cmdLine, "cmdLine");
 
-    ParseResults<CommandSource> results = parse(cmdLine, source, true);
+    String normalizedInput = BrigadierUtils.normalizeInput(cmdLine, true);
+    String[] args = normalizedInput.split(" ");
+    if (dispatcher.getRoot().getChild(args[0]) == null) {
+      return false;
+    }
+    ParseResults<CommandSource> results = dispatcher.parse(normalizedInput, source);
     try {
       return dispatcher.execute(results) != BrigadierCommand.FORWARD;
     } catch (final CommandSyntaxException e) {
-      boolean isSyntaxError = !e.getType().equals(
-              CommandSyntaxException.BUILT_IN_EXCEPTIONS.dispatcherUnknownCommand());
-      if (isSyntaxError) {
-        source.sendMessage(Component.text(e.getMessage(), NamedTextColor.RED));
-        // This is, of course, a lie, but the API will need to change...
-        return true;
-      } else {
-        return false;
-      }
+      source.sendMessage(Component.text(e.getMessage(), NamedTextColor.RED));
+      // This is, of course, a lie, but the API will need to change...
+      return true;
```

Comment:
I agree returning `true` on a syntax exception is reasonable since the API docs say "@return `true` if the command was found and executed". Indeed, the command wasn't executed. Maybe update the API javadoc instead?

Reply:
I'm scared of changing any of the docs so im not touching anything there